components/bms_worker/src/CAN/CANIO-tx.c

```diff
@@ -205,66 +202,129 @@ static const packTable_S* packNextMessage(const packTable_S* packTable,
     return NULL;
 }
 
-static bool MSG_pack_BMS_100Hz(CAN_data_T* message, const uint8_t counter)
+static bool MSG_pack_BMS_100Hz_Critical(CAN_data_T* message, const uint8_t counter)
+{
+    message->u64  = ((uint64_t)((uint16_t)(BMS.voltage.min * 200) & 0x3FF)) << 54;   //10 bits, 5mV precision, range [0-5115]mV
```

Comment:
the way that you've done all these works, however, the more you do this way the more you'll have to **redo** when we implement autogeneration.

See here for how you can do this in such a way that it will easily support conversion to autogeneration:

https://github.com/concordia-fsae/firmware/blob/master/components/steering_wheel/src/CAN/CANIO-tx.c#L69

https://github.com/concordia-fsae/firmware/blob/master/components/steering_wheel/generated/MessagePack.c#L28

https://github.com/concordia-fsae/firmware/blob/master/components/steering_wheel/generated/VEH_sigTx.h#L13

the two in the `generated` dir are the ones that will eventually be autogenerated, and the other one is what we will need to keep doing manually. Note that the autogenerated portion actually encompasses most of the work you've done manually here.

Reply:
I'd love to get on a call with you about this! Will shoot you a message on Teams.

Reply:
I'm not on teams, I'm on the fsae discord though if you're on there